diesel/src/type_impls/date_and_time.rs

```diff
@@ -35,14 +35,16 @@ mod chrono {
         diesel(sql_type = crate::sql_types::Timestamptz)
     )]
     #[cfg_attr(feature = "mysql_backend", diesel(sql_type = crate::sql_types::Datetime))]
+    // TODO: Add TimestamptzSqlite stuff here??
     struct NaiveDateTimeProxy(NaiveDateTime);
 
     #[derive(FromSqlRow)]
-    #[cfg_attr(feature = "postgres_backend", derive(AsExpression))]
     #[diesel(foreign_derive)]
     #[cfg_attr(
         feature = "postgres_backend",
+        derive(AsExpression),
         diesel(sql_type = crate::sql_types::Timestamptz)
     )]
+    #[cfg_attr(feature = "sqlite", derive(AsExpression), diesel(sql_type = crate::sql_types::TimestamptzSqlite))]
```

Comment:
This should be handled in a similar manner than the `AsExpression` derive for `NaiveDateTimeProxy` above. That means the `derive` itself should not be behind a `#[cfg()]` flag as this cases duplicated impls for cases where both backends are enabled due to the fact that rustc then sees the same derive two times.